src/Core/Managed/Shared/Extensibility/Implementation/JsonSerializer.cs

```diff
@@ -54,6 +54,51 @@ public static byte[] Serialize(IEnumerable<ITelemetry> telemetryItems, bool comp
         }
 
         /// <summary>
+        /// Converts serialized telemetry items to a byte array.
+        /// </summary>
+        /// <param name="telemetryItems">Serialized telemetry items.</param>
+        /// <param name="compress">Should serialization also perform compression.</param>
+        /// <returns>The compressed and serialized telemetry items.</returns>
+        [SuppressMessage("Microsoft.Usage", "CA2202:Do not dispose objects multiple times", Justification = "Disposing a MemoryStream multiple times is harmless.")]
+        public static byte[] ConvertToByteArray(string telemetryItems, bool compress = true)
+        {
+            if (string.IsNullOrEmpty(telemetryItems))
+            {
+                throw new ArgumentNullException("telemetryItems");
+            }
+
+            var memoryStream = new MemoryStream();
+            using (Stream compressedStream = compress ? CreateCompressedStream(memoryStream) : memoryStream)
```

Comment:
Gzip steam owns MemoryStream and disposes it: http://stackoverflow.com/questions/8182598/correct-idisposable-implementation-for-this-code